Source/WebKit/UIProcess/Cocoa/UIRemoteObjectRegistry.h

```diff
@@ -26,22 +26,32 @@
 #pragma once
 
 #include "RemoteObjectRegistry.h"
+#include <wtf/RefCounted.h>
 #include <wtf/TZoneMalloc.h>
 #include <wtf/WeakRef.h>
 
 namespace WebKit {
 
 class WebPageProxy;
 
-class UIRemoteObjectRegistry final : public RemoteObjectRegistry {
+class UIRemoteObjectRegistry final : public RemoteObjectRegistry, public RefCounted<UIRemoteObjectRegistry> {
     WTF_MAKE_TZONE_ALLOCATED(UIRemoteObjectRegistry);
 public:
-    UIRemoteObjectRegistry(_WKRemoteObjectRegistry *, WebPageProxy&);
+    static Ref<UIRemoteObjectRegistry> create(_WKRemoteObjectRegistry *remoteObjectRegistry, WebPageProxy& page)
+    {
+        return adoptRef(*new UIRemoteObjectRegistry(remoteObjectRegistry, page));
+    }
+
     ~UIRemoteObjectRegistry();
 
+    void ref() const final { RefCounted::ref(); }
+    void deref() const final { RefCounted::deref(); }
+
     void sendInvocation(const RemoteObjectInvocation&) final;
 
 private:
+    UIRemoteObjectRegistry(_WKRemoteObjectRegistry *, WebPageProxy&);
+
     Ref<WebPageProxy> protectedPage();
     MessageSender messageSender() final;
     uint64_t messageDestinationID() final;
```

Comment:
Well, same old bug. The WeakRef below needs to become a WeakPtr with some null check since you're extending lifetime.

Reply:
Fixed!